### ISSUE TITLE
[Bug] MapInterceptor breaks other interceptors with isArray option

### ISSUE THREAD

**Issue by notmedia** [nartc/mapper#279]
**Is your feature request related to a problem? Please describe.**
Hi! 
I reproduce a bug in this repo - https://github.com/notmedia/automapper-mapinterceptor-bug

First, my case:

1. I have database and Prisma client with generated types
2. I have automapper from PrismaEntity to BackendEntity (using pojos)
3. I have automapper from BackendEntity to EntityDTO (using classes)

When I create two+ resolvers, first for array of entities and the second for one entity - the second interceptor breaks.

```ts
@Resolver(UserDto)
export class UserResolver {
  constructor(private readonly service: AppService) {}

  
  @Query(() => [UserDto])
  @UseInterceptors(
    MapInterceptor(UserDto, User, { isArray: true, mapperName: 'classes' }),
  )
  getUsers() {
    return this.service.getUsers();
  }

  @Query(() => UserDto)
  @UseInterceptors(MapInterceptor(UserDto, User, { mapperName: 'classes' }))
  getUser() {
    return this.service.getUser();
  }
}
```

If you comment `getUsers` resolver than interceptor in next resolver works.
If you put `getUsers` under the `getUser` then `getUser` works, and `getUsers` won't

**Comment by nartc** [comment]
@notmedia I think I have an idea of what it is. Can you please do me a favor and have another `MapInterceptor` but with different models/entities?

**Comment by notmedia** [comment]
@nartc no problem, I tested it and it works with different types. So the bug is in "overlapping" interceptors with same type

**Comment by nartc** [comment]
Yup, which should be a quick fix. Thanks!

**Comment by nartc** [comment]
Fix in 3.4.1 
Please give it a try

**Comment by notmedia** [comment]
@nartc  Yes, it works) Thank you!
